### PR TITLE
Rework word-count documentation

### PR DESCRIPTION
**Pull request by kytrinyx**
Rewriting with the following pretext as the story: frequency counts for optimizing language learning via immersion with dramas and movies.

<details>
<summary>ChatGPT notes</summary>

Counting occurrences of words in a given text can be useful in various real-life scenarios, including:

Content analysis: Researchers and analysts often use word frequency analysis to identify patterns and trends in large volumes of text. This technique is particularly useful for analyzing social media data, news articles, and customer feedback.

Search engine optimization (SEO): To rank well on search engines like Google, website owners often use keywords that are relevant to their content. Counting the frequency of these keywords can help website owners optimize their content and improve their search engine rankings.

Language learning: Language learners often use word frequency lists to identify the most commonly used words in a language. This helps them focus on learning the most important vocabulary first.

Text summarization: In text summarization, the most important information is extracted from a large volume of text. Counting the frequency of words can help identify the most important themes and concepts in the text.

Data mining: Data scientists often use word frequency analysis to mine large volumes of text for insights. This technique can be used to identify patterns in customer feedback, social media posts, and news articles.

-----

There are several different scenarios where frequency counts can be used for language learning, including:

Vocabulary acquisition: Frequency lists can be used to identify the most commonly used words in a language, which can be useful for building vocabulary. Learners can prioritize learning these words to improve their ability to communicate in the language.

Reading comprehension: By identifying the most common words in a text, learners can improve their reading comprehension. If they understand the most common words, they will be able to grasp the overall meaning of the text more easily.

Writing proficiency: Learners can use frequency counts to improve their writing skills. By identifying the most common words in the language, they can ensure that their writing is more natural and easier to understand for native speakers.

Speech production: Frequency lists can help learners identify the most common words and phrases used in conversation. By focusing on these words and phrases, learners can improve their ability to speak the language fluently and naturally.

Accent reduction: Frequency lists can also be used to identify words that are commonly mispronounced by non-native speakers. By focusing on these words, learners can improve their pronunciation and reduce their accent.

Overall, frequency counts can be a valuable tool for language learners in a variety of contexts. By identifying the most commonly used words in a language, learners can focus their efforts on the areas that will have the biggest impact on their ability to communicate effectively in the language.

----

Frequency counts can be used to select TV shows for students to watch by identifying shows that use language that is appropriate for their level of proficiency. Here are some steps you can follow:

Determine the level of proficiency of the students: You can assess the level of proficiency of your students by using standardized language proficiency tests or by evaluating their performance in previous language classes.

Identify the most common words used at their proficiency level: Once you have determined the level of proficiency of your students, you can use frequency lists to identify the most common words used at their level.

Search for TV shows that use these common words: You can search for TV shows that use the common words identified in the frequency lists. This will ensure that the students are exposed to language that is relevant to their level of proficiency.

Check the appropriateness of the content: It's important to make sure that the TV shows you select are appropriate for your students in terms of content. You should consider factors such as age-appropriateness, cultural sensitivity, and relevance to the students' interests.

Preview the shows: Before recommending TV shows to your students, it's a good idea to preview them to make sure they meet your criteria for appropriateness and that the language used is appropriate for your students' level of proficiency.

By using frequency counts to select TV shows for your students to watch, you can ensure that they are exposed to language that is appropriate for their level of proficiency and that will help them improve their language skills.

</details>

### CODE REVIEW

File: exercises/word-count/instructions.md [hunk 1]
```diff
@@ -0,0 +1,73 @@
+# Instructions
+
+Your task is to count how many times each word occurs in a subtitle of a drama.
+
+The dramas that the subtitles are taken from use only ASCII characters.
+
+The characters often speak in casual English, using contractions like _they're_ or _it's_.
+Though these contractions come from two words (e.g. _we are_), the contraction (_we're_) is considered a single word.
+
+Words can be separated by any form of punctuation or whitespace, e.g. "\t", "\n", or " ".
```
Comment (ErikSchierboom) [inline exercises/word-count/instructions.md:10]: Maybe also give some examples of punctuation?

File: exercises/word-count/instructions.md [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+The only punctuation that does not separate words is the apostrophe in contractions.
+
+Numbers are considered words.
+So if the subtitles say `It costs 100 dollars.` then 100 will be its own word.
+However only whole numbers are considered words.
+In the sentence `It costs 24.95.` the words 24 and 95 will be considered separate words.
```
Comment (ErikSchierboom) [inline exercises/word-count/instructions.md:16]: Looking at the canonical data, we don't seem to have this use case. Maybe we can omit this?

**Review by ErikSchierboom — APPROVED** [review]
I like the story!

File: exercises/word-count/instructions.md (hunk 1 above)
Comment (kytrinyx) [reply exercises/word-count/instructions.md:10]: Good call.

File: exercises/word-count/instructions.md (hunk 2 above)
Comment (kytrinyx) [reply exercises/word-count/instructions.md:16]: Ah, right. It follows naturally from the rules provided, but if we don't have the case it doesn't matter. I agree we should just drop it.

File: exercises/word-count/instructions.md (hunk 1 above)
Comment (kytrinyx) [reply exercises/word-count/instructions.md:10]: I think maybe we can just let them discover through failure that punctuation can repeat.

File: exercises/word-count/instructions.md [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+The dramas that the subtitles are taken from use only ASCII characters.
+
+The characters often speak in casual English, using contractions like _they're_ or _it's_.
+Though these contractions come from two words (e.g. _we are_), the contraction (_we're_) is considered a single word.
+
+Words can be separated by any form of punctuation (e.g. ":", "!", or "?") or whitespace (e.g. "\t", "\n", or " ").
+The only punctuation that does not separate words is the apostrophe in contractions.
+
+Numbers are considered words.
+So if the subtitles say `It costs 100 dollars.` then 100 will be its own word.
```
Comment (ErikSchierboom) [inline exercises/word-count/instructions.md:14]: Maybe use quotes here to like with "the word `you`" later on?

```suggestion
So if the subtitles say `It costs 100 dollars.` then `100` will be its own word.
```

**Review by ErikSchierboom — APPROVED** [review]

File: exercises/word-count/instructions.md [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+```text
+scream: 3
+all: 1
+cream: 1
+for: 1
+i: 1
+ice: 1
+we: 1
+you: 1
+```
```
[the remark this answers is not in the document]
Comment (kytrinyx) [reply exercises/word-count/instructions.md:34]: I just copied what was there, mostly :) I'll try a list (I have nothing against lists!)

File: exercises/word-count/instructions.md [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+So if the subtitles say `It costs 100 dollars.` then `100` will be its own word.
+
+Words are case insensitive.
+For example, the word `you` occurs three times in the following sentence:
+
+> You come back, you hear me? DO YOU HEAR ME?
+
+The ordering of the word counts in the results doesn't matter.
+
+The two following groups of counts (for the sentence `I scream, you scream, we all scream for ice cream`) are equivalent:
```
Comment (iHiD) [inline exercises/word-count/instructions.md:23]: I think you can eliminate this example, as it's going to make the instructions very long and I think it doesn't add huge value. The second example below is more useful I think.

File: exercises/word-count/instructions.md [hunk 6]
```diff
@@ -0,0 +1,71 @@
...
+- simple words
+- contractions
+- numbers
+- case insensitive words
+- punctuation (including apostrophes) to separate words
+- different forms of whitespace to separate words
+
+`"That's the password: 'PASSWORD 123'!", cried the Special Agent.\nSo I fled.`
+
+The count for this subtitle would be:
```
Comment (iHiD) [inline exercises/word-count/instructions.md:58]:
```suggestion
The counts for this subtitle would be:
```

(or "mapping" or something?)

Comment (kytrinyx) [reply exercises/word-count/instructions.md:58]: Yeah, maybe mapping.

Comment (kytrinyx) [reply exercises/word-count/instructions.md:58]:
```suggestion
The mapping for this subtitle would be:
```

**Pull request closed without merging**